core-backend/sandbox/src/funcs.rs

```diff
@@ -382,29 +383,54 @@ impl<E: Ext + 'static> FuncsHandler<E> {
         let payload_ptr = pop_i32(&mut args)?;
         let payload_len = pop_i32(&mut args)?;
         let value_ptr = pop_i32(&mut args)?;
+        let message_id_ptr = pop_i32(&mut args)?;
 
-        let result = ctx
-            .ext
+        ctx.ext
+            .clone()
             .with(|ext| {
                 let payload = funcs::get_vec(&ctx.memory, payload_ptr, payload_len)?;
                 let value = funcs::get_u128(&ctx.memory, value_ptr)?;
                 ext.reply(ReplyPacket::new(payload, value))
-                    .map(|_| ())
-                    .into_error_code()
+                    .on_success_code(|message_id| wto(ctx, message_id_ptr, message_id.as_ref()))
             })
+            .map_err(Into::into)
             .and_then(|res| res.map(Value::I32).map(ReturnValue::Value))
             .map_err(|_| {
-                ctx.trap = Some("Trapping: unable to send reply message");
+                ctx.trap = Some("Trapping: unable to send message");
```

Comment:
Why does this trap explanation changed?

```suggestion
                ctx.trap = Some("Trapping: unable to send reply message");
```